control-base/src/main/java/fi/nls/oskari/control/data/CreateUserLayerHandler.java

```
@@ -115,17 +116,22 @@ public void handleAction(ActionParameters params) throws ActionException {
             response.setCharacterEncoding("UTF-8");
 
             JSONObject userLayer = userlayerService.parseUserLayer2JSON(ulayer);
+            JSONHelper.putValue(userLayer, "featuresCount", ulayer.getFeatures_count());
             JSONObject permissions = OskariLayerWorker.getAllowedPermissions();
             JSONHelper.putValue(userLayer, "permissions", permissions);
+            //add warning if features were skipped
+            if (ulayer.getFeatures_skipped() > 0) {
+                JSONObject featuresSkipped = new JSONObject();
+                JSONHelper.putValue(featuresSkipped, "featuresSkipped", ulayer.getFeatures_skipped());
+                JSONHelper.putValue(userLayer, "warning", featuresSkipped);
+            }
             response.getWriter().print(userLayer);
 
         }catch (ActionException e) {
-            throw new ActionException("Import failed - " + e.getMessage(),
-                    e);
+            throw new ActionException (e.getMessage());
```

Maybe just rethrow the original exception?